src/placeos-triggers/loki_search_errrors.cr

```diff
@@ -0,0 +1,53 @@
+require "time"
+require "placeos-models"
+require "tasker"
+require "loki-client"
+
+module PlaceOS::Triggers
+  class LokiSearchForErrors
+    Log = ::Log.for(self)
+
+    def initialize(@repeat_interval : Time::Span)
+      @client = Loki::Client.from_env
+    end
+
+    def self.new(interval : String)
+      new(Triggers.extract_time_span(interval))
+    end
+
+    def start
+      labels = @client.list_labels.data
+      stream = labels.try &.includes?("container") ? "container" : "app"
+      query = %({#{stream}="core"} |~ "mod-*" |~ "(?i)exception" | logfmt | level = "[E]")
+
+      Tasker.every(@repeat_interval) do
+        Log.debug { "Searching Loki for runtime error logs of all running modules" }
+        PlaceOS::Model::Module.where(running: true).each do |mod|
+          Log.debug { {message: "Searching Loki for module errors", module: mod.id, driver: mod.driver_id} }
+          begin
+            errors = Array(Tuple(String, String)).new
+            result = @client.query_range(query, 1000, Time.utc - 24.hour, Time.utc, Loki::Direction::Backward)
+            result.response_data.result.as(Loki::Model::Streams).each do |stream|
```

Comment:
Shadowing outer local variable `stream`
I think it's important not to shadow the variable names, maybe call the outer variable `stream_label`?